### ISSUE TITLE
Error: PHP configuration mode fpm was selected, but is not supported on this system! on FreeBSD 11.1

### ISSUE THREAD

**Issue by jwhulette** [virtualmin/virtualmin-gpl#51]
When creating a new virtual server I get the error. I am able to create the proper apache configuration and pool manually. Once created manually it works fine.

**Comment by Vagrantin** [comment]
Can you provide a bit more details about the version you are running ?
Virtualmin
PHP
Apache
Do you have the proper Apache module ?
Could it be a path issue as well ?

thinking out-loud

hope that help

Matth

**Comment by jwhulette** [comment]
Virtualmin:   6.02
Webmin: 1.880
Apache: 2.4.29 
PHP: 7.1.14
FreeBSD: 11.1-RELEASE-p6

It could be a path issue, I have run into that before on FreeBSD. But the error just says it's not supported. I looked into the code, but my Perl foo is not that great.

**Comment by jcameron** [comment]
Where are the PHP-FPM config files on your system? Virtualmin may not know where to look on FreeBSD

**Comment by jwhulette** [comment]
The path is /usr/local/etc/php-fpm.d

**Comment by jcameron** [comment]
Thanks! What's the name of the RC script or equivalent that starts the FPM server on FreeBSD though?

**Comment by jwhulette** [comment]
/usr/local/etc/rc.d/php-fpm

**Comment by jacobsenw** [comment]
I recently ran into this issue as well.
To get around it I updated the list of php fpm packages the code was checking for to include "php7" as both the php, php-cgi and php-fpm are installed via this one package on FreeBSD.

Just as an fyi for anyone else, you will also need to "adjust" the code that writes the php-fpm conf files for domains to include the 127.0.0.1 ip address as well as the port number when defining the "listen" parameter.
